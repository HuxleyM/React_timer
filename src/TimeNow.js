import React, { Component } from 'react';

class TimeNow extends Component {
    constructor(){
        super()
        this.state = {
            time : this.generateTime()
        }
    }

    generateTime(){
        const time = new Date();
        const hour = time.getHours();
        const mins = time.getMinutes();
        const sec = time.getSeconds();
        return `${hour}:${mins}:${sec}`
    }

    componentDidMount(){
        this.timer = setInterval( ()=> this.tick(), 1000 );   
    }

    componentWillUnmount(){
       clearInterval(this.timer)
    }

    tick(){
        this.setState({ time : this.generateTime() })
    }

    render() {
        return (
            <div id='time_now'>
            <h2>{this.state.time}</h2>
            </div>
        )
    }
}

 export default TimeNow