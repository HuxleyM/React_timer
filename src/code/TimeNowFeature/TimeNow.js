import React, { Component } from 'react';

class TimeNow extends Component {
    constructor(){
        super()
        this.state = {
            time : this.generateTime()
        }
    }

    generateTime(){
        let time = new Date();
        return `${time.getHours()}:${time.getMinutes()}:${time.getSeconds()}`
    }

    componentDidMount(){
        this.timer = setInterval( ()=> this.tick(), 1000 );   
    }

    componentWillUnmount(){
       clearInterval(this.timer)
    }

    tick(){
        this.setState({ time : this.generateTime() })
    }

    render() {
        return <h2>{this.state.time}</h2>
    }
}

 export default TimeNow