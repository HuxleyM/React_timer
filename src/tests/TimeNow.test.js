import React from 'react'
import TimeNow from '../code/TimeNowFeature/TimeNow'
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme'

describe('clicked on clock', ()=>{
 
    const wrap = mount( <TimeNow/>) 
    const currentDate = new Date()  
    
    it('will generate correct time_now  ', ()=>{
        //setup
        const wrap = shallow( <TimeNow/>)  
        expect(wrap.instance().generateTime()).toBe(`${currentDate.getHours()}:${currentDate.getMinutes()}:${currentDate.getSeconds()}`)
    })

    it('should disply this time on the screen', ()=>{
         const value = wrap.find('#time_now h2').text()
         expect(value).toBe(`${currentDate.getHours()}:${currentDate.getMinutes()}:${currentDate.getSeconds()}`)
    })

    it('should tick', ()=>{
        // is need a time wizard
       jest.useFakeTimers(); 
       const wrap = mount( <TimeNow/>) 
       expect(setInterval).toHaveBeenCalledTimes(1);
    })

    it('display should change per second', ()=>{
        // need to do more snap shot testing
        // const tree = renderer.create(<TimeNow />).toJSON();
        // expect(tree).toMatchSnapshot();
        // var secondV=  setTimeout(function() {const secondV = renderer.create(<TimeNow/>).toJSON(); return secondV; }, 1000)
        // expect(secondV).toMatchSnapshot();

    })
})