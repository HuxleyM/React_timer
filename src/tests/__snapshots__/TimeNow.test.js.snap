// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clicked on clock display should change per second 1`] = `3`;

exports[`clicked on clock display should change per second 2`] = `3`;
